eclair-core/src/main/scala/fr/acinq/eclair/json/JsonSerializers.scala

```diff
@@ -468,22 +468,20 @@ object CustomTypeHints {
     classOf[MessageReceivedJson] -> "onion-message-received"
   ))
 
-  val channelStates: ShortTypeHints = ShortTypeHints(
-    List(
-      classOf[Nothing],
-      classOf[DATA_WAIT_FOR_OPEN_CHANNEL],
-      classOf[DATA_WAIT_FOR_ACCEPT_CHANNEL],
-      classOf[DATA_WAIT_FOR_FUNDING_INTERNAL],
-      classOf[DATA_WAIT_FOR_FUNDING_CREATED],
-      classOf[DATA_WAIT_FOR_FUNDING_SIGNED],
-      classOf[DATA_WAIT_FOR_FUNDING_LOCKED],
-      classOf[DATA_WAIT_FOR_FUNDING_CONFIRMED],
-      classOf[DATA_NORMAL],
-      classOf[DATA_SHUTDOWN],
-      classOf[DATA_NEGOTIATING],
-      classOf[DATA_CLOSING],
-      classOf[DATA_WAIT_FOR_REMOTE_PUBLISH_FUTURE_COMMITMENT]
-    ), typeHintFieldName = "type")
+  val channelData: CustomTypeHints = CustomTypeHints(Map(
+    classOf[ChannelData.WaitingForOpenChannel] -> "waiting-for-open-channel",
+    classOf[ChannelData.WaitingForAcceptChannel] -> "waiting-for-accept-channel",
+    classOf[ChannelData.WaitingForFundingInternal] -> "waiting-for-funding",
+    classOf[ChannelData.WaitingForFundingCreated] -> "waiting-for-funding-created",
+    classOf[ChannelData.WaitingForFundingSigned] -> "waiting-for-funding-signed",
+    classOf[ChannelData.WaitingForFundingConfirmed] -> "waiting-for-funding-confirmed",
+    classOf[ChannelData.WaitingForFundingLocked] -> "waiting-for-funding-locked",
+    classOf[ChannelData.Normal] -> "normal",
+    classOf[ChannelData.ShuttingDown] -> "shutting-down",
+    classOf[ChannelData.Negotiating] -> "negotiating-closing-fees",
+    classOf[ChannelData.Closing] -> "closing",
+    classOf[ChannelData.WaitingForRemotePublishFutureCommitment] -> "waiting-for-remote-publish-future-commitment",
+  ))
```

Comment:
My thought was that if some clients use this to figure out what data to expect and programmatically access, we should provide something stable. Directly using the type name wouldn't be stable (if we change the type name, we would inadvertently change what was in the JSON).

But I agree this is not very future-proof because of the matching, and probably not useful for now, so we can remove it.